### PR TITLE
Implement "RecordMarker", "StartTimer", "CancelTimer" and "CancelWorkflowExecution" decisions

### PR DESCRIPTION
**Pull request by JimNero009**
This PR somewhat addresses https://github.com/spulec/moto/issues/4708, in that more of the unimplemented decision types now have logic backing them.

I was interested also in the StartChildWorkflowExecution decision, but felt the PR was getting big enough and I can live without that for a little while longer.

I have also tried to implement a couple of 'common' failure modes for these decisions, but they are not completed. I have marked TODOs in the code where other failure states should be considered.

### CODE REVIEW

File: tests/test_swf/models/test_workflow_execution.py [hunk 1]
```diff
@@ -580,8 +579,52 @@ def test_create_timer():
         last_event.event_attributes["decisionTaskCompletedEventId"].should.equal(123)
 
 
+def test_start_timer_correctly_fires_timer_later():
+    wfe = make_workflow_execution()
+    START_TIMER_EVENT_ATTRIBUTES = {"startToFireTimeout": "60", "timerId": "abc123"}
+
+    class MockEvent:
```
Comment (JimNero009) [inline tests/test_swf/models/test_workflow_execution.py:586]: Note to self -- probably only actually need to mock the 'wait' function.

**Review by bblommers — COMMENTED** [review]
LGTM so far @JimNero009! It would be good to have an integration test as well, to verify the actual behaviour of the timer - so without any Mocks involved.

There are a few tests already in `responses/test_workflow_executions.py` that you can borrow from. (Note that you'd only need to write a test for `mock_swf` - no need to verify the behaviour under the deprecated decorator.)

If a 'perfect' integration test means adding some `time.sleep()` statements, that's OK as well.

**Comment by JimNero009** [comment]
Hi Bert, thanks for the suggestion -- I was thinking myself that essentially inserting a full workflow definition to follow that includes these events would be great, so awesome to see you have a suite for that.

Thanks for the eyes, good to know I'm going in generally the right direction, will keep at it!

**Comment by JimNero009** [comment]
I don't _think_ that test failure is caused by my change?

**Comment by bblommers** [comment]
> I don't _think_ that test failure is caused by my change?

No, that was an intermittent failure - I've restarted the build and it all looks OK now

**Review by bblommers — APPROVED** [review]
Approved - thanks for the PR @JimNero009!

**Pull request merged**